### PR TITLE
Add support for S3 endpoint-url (for S3-compatible storage solutions)

### PR DESCRIPTION
**Pull request by kjothen**
* new optional input and output task parameter :s3/endpoint-url
* upgraded aws sdk from 1.11.76 to 1.11.125
* changed s3-utils/new-client to align with current recommendations to create AmazonS3Client objects
* removed s3-utils fns to set region, endpoint, etc.
* added a build.gradle file for, ahem, gradle builds

### CODE REVIEW

**Comment by MichaelDrogalis** [comment]
Thanks! We'll get a look at this today. cc @lbradstreet

**Comment by kjothen** [comment]
@MichaelDrogalis @lbradstreet Thanks! I was initially thrown by the fact that the end-to-end test fails. I don't have access to the existing "s3-plugin-test-onyx" bucket, and the test attempts to create it each time before using it. Under my AWS account this fails at the .createBucket call with the exception "requested bucket name is not available". Under your account, it silently passes over the fact you've created the bucket already, and the tests work. 

TL;DR - please consider creating a random bucket in the end-to-end test, mirroring what the output-test does (I should have committed this change!)

**Comment by lbradstreet** [comment]
Thanks @kjothen! Looks like a good PR!

Yeah, that's a bit of a tough one with the bucket! I tried to avoid creating a random bucket as these things have a way of not being cleaned up when tests fail, especially on CI. I'll think about it, otherwise maybe I'll make the warning better and the bucket more configurable.

**Comment by lbradstreet** [comment]
@kjothen could you comment further on the recommendation for creating S3 client objects? We tried to avoid forcing users to provide their S3 credentials via the task-map, as many users will not want to store their S3 credentials in ZooKeeper (via Onyx's job submission). Therefore, we want to at least support the default provider chain which will attempt to use the standard environment variables and standard AWS config paths.

**Comment by lbradstreet** [comment]
Apologies, I looked at the S3 builder docs and I see it defaults to that provider chain. Good stuff!

**Comment by kjothen** [comment]
@lbradstreet thanks for persevering with this, but to reassure you, I've tested the plugin both with supplying credentials and without (using env cars or AWS configuration only)

**Comment by lbradstreet** [comment]
Great! Thanks.

**Comment by lbradstreet** [comment]
Done. Thanks for the great PR!

**Comment by lbradstreet** [comment]
Released in 0.10.0.0-beta17.
